interfaces/cos_agent/v0/schema.py

```diff
@@ -0,0 +1,140 @@
+                                    "app_group": "cloud",
+                                    "juju_application": "zookeeper",
+                                    "juju_charm": "zookeeper",
+                                    "juju_model": "test",
+                                    "juju_model_uuid": "58b48ff0-608b-435f-83b9-ea643e0b98db",
+                                    "severity": "critical"
+                                }
+                            }
+                            ],
+                        },
+                        ],
+                },
+                "log_alert_rules": {},
+                "dashboards": [
+                    "/Td6WFoAAATm1rRGAgAhARYAAAB0L+Wj6hvVPXVdA…",
+                    "/Td6WFoAAATm1rRGAgAhARYAAAB0L+Wj6hvVPXVdB…",
+                ],
+                "log_slots": ["charmed-zookeeper:logs"]
+            }
+    RequirerSchema:
+        unit: <empty>
+        app: <empty>
+"""
+
+import base64
+import json
+import lzma
+from pydantic import BaseModel
+from typing import ClassVar, Dict, List, Union, Optional
+from interface_tester.schema_base import DataBagSchema
+
+
+class GrafanaDashboard(str):
+    """Grafana Dashboard encoded json; lzma-compressed."""
+
+    # TODO Replace this with a custom type when pydantic v2 released (end of 2023 Q1?)
+    # https://github.com/pydantic/pydantic/issues/4887
+    @staticmethod
+    def _serialize(raw_json: Union[str, bytes]) -> "GrafanaDashboard":
+        if not isinstance(raw_json, bytes):
+            raw_json = raw_json.encode("utf-8")
+        encoded = base64.b64encode(lzma.compress(raw_json)).decode("utf-8")
+        return GrafanaDashboard(encoded)
+
+    def _deserialize(self) -> Dict:
+        raw = lzma.decompress(base64.b64decode(self.encode("utf-8"))).decode()
+        return json.loads(raw)
+
+    def __repr__(self):
+        """Return string representation of self."""
+        return "<GrafanaDashboard>"
+
+
+class CosAgentProviderUnitData(BaseModel):
+    """Unit databag model for `cos-agent` relation."""
+
+    # The following entries are the same for all units of the same principal.
+    # Note that the same grafana agent subordinate may be related to several apps.
+    # this needs to make its way to the gagent leader
+    metrics_alert_rules: dict
+    log_alert_rules: dict
+    dashboards: List[GrafanaDashboard]
+
+    # The following entries may vary across units of the same principal app.
+    # this data does not need to be forwarded to the gagent leader
+    metrics_scrape_jobs: List[Dict]
+    log_slots: List[str]
+
+    # when this whole datastructure is dumped into a databag, it will be nested under this key.
+    # while not strictly necessary (we could have it 'flattened out' into the databag),
+    # this simplifies working with the model.
+    KEY: ClassVar[str] = "config"
+
+
+class CosAgentPeersUnitData(BaseModel):
+    """Unit databag model for `cluster` cos-agent machine charm peer relation."""
```

Comment:
the name of the relation endpoint we use, or the fact that it's a machine charm, are not relevant here. Nor that it is a peer relation.